landoapi/models/landing_job.py

```diff
@@ -166,9 +169,26 @@ def serialized_landing_path(self):
             ]
 
     @property
-    def head_revision(self) -> str:
+    def landing_job_identifier(self) -> str:
         """Human-readable representation of the branch head's Phabricator revision ID."""
-        return f"D{self.revisions[-1].revision_id}"
+        if not self.revisions:
+            raise ValueError(
+                "Job must be associated with a revision to have a head revision."
+            )
+
+        head = self.revisions[-1]
+        if head.revision_id:
+            # Return the Phabricator identifier if the head revision has one.
+            return f"D{self.revisions[-1].revision_id}"
+
+        commit_message = head.patch_data.get("commit_message")
+        if commit_message:
+            # If there is no Phabricator identifier, return the first line of the
+            # commit message for the patch.
+            return commit_message.splitlines()[0]
+
+        # Return a placeholder in the event neither exists.
+        return "unknown"
```

Comment:
There should always be at least a commit hash, so we should return that in case nothing else is available (though that can be covered by the previous statement).